consensus/istanbul/backend/backend.go

```diff
@@ -432,6 +432,18 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		return errInvalidProposal
 	}
 
+	// If the current block occurred before the Celo1 hard fork, check that the author and coinbase are equal.
+	if !sb.chain.Config().IsCelo1(block.Number()) {
```

Comment:
wrong place to add this check. We don't need to add a check but to prevent generating new blocks with Coinbase != Validator, since old versions assume they are the same.